java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -27,6 +28,7 @@
 public class ScanOptions {
   private final Optional<String[]> columns;
   private final long batchSize;
+  private ByteBuffer substraitExtendedExpression;
```

Comment:
This must be final.

Reply:
This is optional, is up to the user to configure this or not, it is part of the builder option